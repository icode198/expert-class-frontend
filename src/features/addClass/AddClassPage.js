import React, { useState } from 'react';
import axios from 'axios';
import getCSRFToken from '../../app/getCSRFToken';
import expertClassAPI from '../../app/expertClassAPI';

const AddClassPage = () => {
  const handleSubmit = async (event) => {
    event.preventDefault();
    const formData = new FormData(event.target);

    await axios.post('http://localhost:3001/api/v1/courses',
      formData,
      {
        withCredentials: true,
        headers: {
          'X-CSRF-Token': getCSRFToken(),
          'content-type': 'multipart/form-data',
        },
      });
  };

  const [classes, setClasses] = useState(null);

  const [course, setCourse] = useState(null);

  const classesList = (courses) => courses.map((c) => (
    <div key={c.id}>
      <img src={c.course_image_url} alt="class instructor" />
      <h3>{c.title}</h3>
      <p>{c.description}</p>
    </div>
  ));

  const showClass = (c) => (
    <div key={c.id}>
      <img src={c.course_image_url} alt="class instructor" />
      <h3>{c.title}</h3>
      <p>{c.description}</p>
    </div>
  );

  const fetchClasses = async () => {
    const response = await expertClassAPI.get('/api/v1/courses');
    setClasses(classesList(response.data));
  };

  const fetchClass = async () => {
    const response = await expertClassAPI.get('/api/v1/courses/16');
    setCourse(showClass(response.data));
  };

  return (
    <>
      <div style={{ marginLeft: 400 }}>
        <h1>Create a New Class</h1>
        <form onSubmit={handleSubmit}>
          <label htmlFor="title">
            Title
            <input type="text" name="course[title]" />
          </label>
          <label htmlFor="description">
            Description
            <input type="text" name="course[description]" />
          </label>
          <label htmlFor="duration">
            Duration
            <input type="number" name="course[duration]" />
          </label>
          <label htmlFor="instructor">
            Instructor
            <input type="text" name="course[instructor]" />
          </label>
          <label htmlFor="image">
            Upload image of Instructor
            <input type="file" name="course[image]" accept="image/png, image/jpeg" />
          </label>
          <input type="submit" value="Upload" />
        </form>
      </div>

      <div style={{
        width: 600, marginLeft: 400, marginTop: 100, minHeight: 200,
      }}
      >
        <button type="button" onClick={fetchClasses}>Fetch Classes</button>
        {classes}
      </div>
      <div style={{
        width: 600, marginLeft: 400, marginTop: 100, minHeight: 200,
      }}
      >
        <button type="button" onClick={fetchClass}>Fetch One Class</button>
        {course}
      </div>
    </>
  );
};

export default AddClassPage;
