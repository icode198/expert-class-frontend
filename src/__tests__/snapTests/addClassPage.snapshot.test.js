import React from 'react';
import { cleanup } from '@testing-library/react';
import { renderWithRedux } from '../../tests/test-utils';
import AddClassPage from '../../pages/addClassPage/addClassPage';

afterEach(cleanup);

describe('AddClassPage', () => {
  test('Renders the page', async () => {
    const app = renderWithRedux(<AddClassPage />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
