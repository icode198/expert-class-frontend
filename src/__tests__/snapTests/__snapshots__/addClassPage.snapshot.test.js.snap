// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddClassPage Renders the page 1`] = `
<div
  className="add-class-container"
>
  <div
    className="main-container"
  >
    <h2
      className="heading"
    >
      Add Class
    </h2>
    <form
      className="add-class-form"
      onSubmit={[Function]}
    >
      <div
        className="inputWrapper"
      >
        <label
          htmlFor="title"
        >
          <input
            className="add-class-title"
            id="title"
            name="course[title]"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        className="inputWrapper"
      >
        <label
          htmlFor="description"
        >
          <input
            className="class-description"
            id="description"
            name="course[description]"
            onChange={[Function]}
            placeholder="Description"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        className="inputWrapper"
      >
        <label
          htmlFor="duration"
        >
          <input
            className="class-duration"
            id="duration"
            name="course[duration]"
            onChange={[Function]}
            placeholder="Duration"
            type="number"
            value=""
          />
        </label>
      </div>
      <div
        className="inputWrapper"
      >
        <label
          htmlFor="instructor"
        >
          <input
            className="class-instructor"
            id="instructor"
            name="course[instructor]"
            onChange={[Function]}
            placeholder="Instructor"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        className="upload-box inputWrapper"
      >
        <span
          className="instruction"
        >
          Upload Instructor Image
        </span>
        <input
          accept="image/png, image/jpeg"
          className="instructor-image"
          id="image"
          name="course[image]"
          onChange={[Function]}
          placeholder="Add image"
          type="file"
        />
      </div>
      <div>
        <input
          className="class-submit"
          type="submit"
          value="Upload"
        />
      </div>
    </form>
  </div>
</div>
`;
